### PR TITLE
Make resource definition functions public

### PR DESCRIPTION
**Pull request by zli82016**
We are going to create a new package provider and move provider.go to this package.
In provider.go file, the resource definition functions in google package are called. The provider.go file will be moved to a new package provider. The resource definition functions need to be public by capitalizing the function names.

Also, some utility functions called in provider.go file need to be public.

multiEnvSearch
expandProviderBatchingConfig
configureDCLProvider
providerDCLConfigure

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: mmv1/third_party/terraform/utils/provider.go.erb [hunk 1]
```diff
@@ -191,154 +191,154 @@ func Provider() *schema.Provider {
...
 			<% end -%>
-			"google_folder":                                    dataSourceGoogleFolder(),
-			"google_folders":                                   dataSourceGoogleFolders(),
-			"google_folder_organization_policy":                dataSourceGoogleFolderOrganizationPolicy(),
-			"google_logging_project_cmek_settings":             dataSourceGoogleLoggingProjectCmekSettings(),
+			"google_folder":                                    DataSourceGoogleFolder(),
+			"google_folders":                                   DataSourceGoogleFolders(),
+			"google_folder_organization_policy":                DataSourceGoogleFolderOrganizationPolicy(),
+			"google_logging_project_cmek_settings":             DataSourceGoogleLoggingProjectCmekSettings(),
 			"google_logging_sink":                              dataSourceGoogleLoggingSink(),
```
Comment (rileykarson) [inline mmv1/third_party/terraform/utils/provider.go.erb:294]: nit: This one seems to have been missed

File: mmv1/third_party/terraform/utils/provider.go.erb [hunk 2]
```diff
@@ -191,154 +191,154 @@ func Provider() *schema.Provider {
...
+			"google_service_account":                           DataSourceGoogleServiceAccount(),
+			"google_service_account_access_token":              DataSourceGoogleServiceAccountAccessToken(),
+			"google_service_account_id_token":                  DataSourceGoogleServiceAccountIdToken(),
+			"google_service_account_jwt":                       DataSourceGoogleServiceAccountJwt(),
+			"google_service_account_key":                       DataSourceGoogleServiceAccountKey(),
+			"google_sourcerepo_repository":                     DataSourceGoogleSourceRepoRepository(),
+			"google_spanner_instance":                          DataSourceSpannerInstance(),
+			"google_sql_ca_certs":                              DataSourceGoogleSQLCaCerts(),
+			"google_sql_backup_run":                            DataSourceSqlBackupRun(),
 			"google_sql_databases":                             dataSourceSqlDatabases(),
```
Comment (rileykarson) [inline mmv1/third_party/terraform/utils/provider.go.erb:326]: nit: This one seems to have been missed

**Review by rileykarson — APPROVED** [review]
Two functions snuck through, otherwise LGTM

File: mmv1/third_party/terraform/utils/provider.go.erb (hunk 1 above)
Comment (zli82016) [reply mmv1/third_party/terraform/utils/provider.go.erb:294]: Good catch. Thanks.

File: mmv1/third_party/terraform/utils/provider.go.erb (hunk 2 above)
Comment (zli82016) [reply mmv1/third_party/terraform/utils/provider.go.erb:326]: Good catch. Thanks.

**Pull request merged**